src/tokenRefresh.ts

```diff
@@ -1,41 +1,41 @@
 import apiClient from "./apiClient"
-import { Service, postData } from "./base"
-
-const refreshToken = async () => {
-  const { refresh_token, access_token } = await postData(
-    Service.TOKEN_REFRESH,
-    "refresh-token",
-    { token: apiClient.tokens.refreshToken }
-  )
-
-  apiClient.setTokens({
-    accessToken: access_token,
-    refreshToken: refresh_token,
-  })
+
+const invokeTokenRefresh = async () => {
+  const { accessToken } = await apiClient.refreshToken()
+
+  apiClient.setAccessToken(accessToken)
   if (apiClient.handlers.onAccessTokenUpdate) {
-    apiClient.handlers.onAccessTokenUpdate(apiClient.tokens.accessToken)
+    apiClient.handlers.onAccessTokenUpdate(apiClient.accessToken)
```

Comment:
is this something the following code needs to wait for to prevent race conditions? like... the token is updated in the client, but not yet via the callback in the project

Reply:
Not sure if I understand you correctly this is the client callback executor

Reply:
Ahh gotcha in the other comment will check it